package.json

```diff
@@ -47,14 +47,23 @@
     "@types/node": "^22.7.5",
     "eslint": "^9.14.0",
     "eslint-config-prettier": "^9.1.0",
+    "husky": "^9.1.6",
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
+    "lint-staged": "^15.2.10",
     "prettier": "^3.3.3",
     "ts-jest": "^29.2.5",
     "typescript": "^5.6.3",
     "typescript-eslint": "^8.12.2"
   },
   "engines": {
     "node": ">=18"
+  },
+  "lint-staged": {
+    "*.{js,ts}": [
+      "prettier --write",
+      "eslint --fix"
```

Comment:
Wouldn't it make sense to just run eslint since it should already run through prettier? Since eslint also runs prettier (through the plugin we added)?

Reply:
> Wouldn't it make sense to just run eslint since it should already run through prettier? Since eslint also runs prettier (through the plugin we added)?

This actually does not happen, the config that is included from `eslint-config-prettier` does not integrate the tools together, it only disables rules in ESLint that could possibly cause conflicts with Prettier to avoid race conditions.